client/my-sites/upgrades/domain-management/transfer/pending-transfer.jsx

```diff
@@ -0,0 +1,125 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import SectionHeader from 'components/section-header';
+import Button from 'components/button';
+import { acceptTransfer, declineTransfer } from 'lib/upgrades/actions';
+import notices from 'notices';
+import { translate } from 'lib/mixins/i18n';
+
+const somethingWrongMessage = translate(
+	'Oops! Something went wrong and your request could not be ' +
+	'processed. Please try again or {{a}}Contact Support{{/a}} if ' +
+	'you continue to have trouble.',
+	{
+		components: {
+			a: (
+				<a
+					href="https://support.wordpress.com/contact/"
+					target="_blank" />
+			)
+		}
+	}
+);
+
+const Unlocked = React.createClass( {
+	getInitialState() {
+		return {
+			submitting: false
+		}
+	},
+
+	handleAcceptTransferClick() {
+		this.setState( { submitting: true } );
+
+		acceptTransfer( this.props.selectedDomainName, ( error ) => {
+			this.setState( { submitting: false } );
+			this.displayAcceptTransferNotice( error );
+		} );
+	},
+
+	displayAcceptTransferNotice( error ) {
+		if ( error ) {
+			notices.error( somethingWrongMessage );
+
+			return;
+		}
+
+		notices.success(
+			this.translate(
+				"Sorry to see you go! We've accepted the domain transfer, you " +
+				'should receive an email confirming this from your new registrar ' +
+				'shortly.'
+			)
+		);
+	},
+
+	handleDeclineTransferClick() {
+		this.setState( { submitting: true } );
+
+		declineTransfer( this.props.selectedDomainName, ( error ) => {
+			this.setState( { submitting: false } );
+			this.displayDeclineTransferNotice( error );
+		} );
+	},
+
+	displayDeclineTransferNotice( error ) {
+		if ( error ) {
+			notices.error( somethingWrongMessage );
+
+			return;
+		}
+
+		notices.success(
+			this.translate(
+				"Glad you want to stay! We've declined the domain transfer request."
+			)
+		);
+	},
+
+	render() {
+		return (
+			<div>
+				<SectionHeader label={ this.translate( 'Transfer Domain' ) } className="transfer__section-header"/>
+				<Card className="transfer-card">
+					<p>
+						{ this.translate(
+							'Your domain %(domainName)s, is currently being transferred to another registrar. To speed up this ' +
+							'process, confirm the transfer below. {{learnMoreLink}}Learn more.{{/learnMoreLink}}',
```

Comment:
Maybe we should use the same wording as on the button (or the other way around). Either the user should accept the transfer or confirm it - less confusing IMHO.